src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,106 @@
+    private final ArmIO armIO = ArmIO.generateIO();
+    private final ArmInputsAutoLogged armInputs = new ArmInputsAutoLogged();
+
+    private TrapezoidProfile
+            angleMotorProfile = null,
+            elevatorMotorProfile = null;
+    private double
+            lastAngleMotorProfileGenerationTime,
+            lastElevatorMotorProfileGenerationTime;
+
+    public static Arm getInstance() {
+        return INSTANCE;
+    }
+
+    private Arm() {
+    }
+
+    @Override
+    public void periodic() {
+        armIO.updateInputs(armInputs);
+        Logger.processInputs("Arm", armInputs);
+        updateMechanism();
+    }
+
+    void generateAngleMotorProfile(Rotation2d targetAngle, double speedPercentage) {
+        angleMotorProfile = new TrapezoidProfile(
+                Conversions.scaleConstraints(ArmConstants.ANGLE_CONSTRAINTS, speedPercentage),
+                new TrapezoidProfile.State(targetAngle.getDegrees(), 0),
+                new TrapezoidProfile.State(armInputs.anglePositionDegrees, armInputs.angleVelocityDegreesPerSecond)
+        );
+        lastAngleMotorProfileGenerationTime = Timer.getFPGATimestamp();
+    }
+
+    void generateElevatorMotorProfile(double targetElevatorPositionMeters, double speedPercentage) {
+        elevatorMotorProfile = new TrapezoidProfile(
+                Conversions.scaleConstraints(ArmConstants.ELEVATOR_CONSTRAINTS, speedPercentage),
+                new TrapezoidProfile.State(targetElevatorPositionMeters, 0),
+                new TrapezoidProfile.State(armInputs.elevatorPositionMeters, armInputs.elevatorVelocityMetersPerSecond)
+        );
+        lastElevatorMotorProfileGenerationTime = Timer.getFPGATimestamp();
+    }
+
+    void setTargetAngleFromProfile() {
+        if (angleMotorProfile == null) {
+            armIO.stopAngleMotors();
+            return;
+        }
+
+        TrapezoidProfile.State targetState = angleMotorProfile.calculate(getAngleProfileTime());
+        ArmConstants.TARGET_ARM_POSITION_LIGAMENT.setAngle(targetState.position);
+        armIO.setTargetAngleState(targetState);
+    }
+
+    void setTargetElevatorPositionFromProfile() {
+        if (elevatorMotorProfile == null) {
+            armIO.stopElevatorMotors();
+            return;
+        }
+
+        TrapezoidProfile.State targetState = elevatorMotorProfile.calculate(getElevatorProfileTime());
+        ArmConstants.TARGET_ARM_POSITION_LIGAMENT.setLength(targetState.position + ArmConstants.RETRACTED_ARM_LENGTH_METERS);
+        armIO.setTargetElevatorState(targetState);
+    }
+
+    boolean isElevatorRising(double targetElevatorPositionMeters) {
+        return targetElevatorPositionMeters >= getElevatorPositionMeters();
+    }
+
+    boolean atAngle(Rotation2d targetAngle) {
+        return Math.abs(armInputs.anglePositionDegrees - targetAngle.getDegrees()) <= ArmConstants.ANGLE_TOLERANCE_DEGREES;
+    }
+
+    boolean atElevatorPositionMeters(double targetElevatorPositionMeters) {
```

Comment:
for the function name, you can call it `atElevatorPosition` and let the variable explain the units